### DISCUSSION TITLE
Possible Bug in Postprocessors::SideFluxIntegral

### DISCUSSION THREAD

**Discussion by wrkendrick** [idaholab/moose#17608] [Q&A General] [answered]
Hi all, I've been running some heat conduction in a very simple geometry focused at returning heat fluxes on a surface. I recently updated to the current Master branch of MOOSE (I was previously on a Master checkpoint from around January) and after that my heat fluxes have had a bug. 

After the initial construction, on the very first non-linear iteration, the heat fluxes are reported in the ~30kW range, when they should be basically zero. This is correct by the next non-linear iteration, and is correct in the initial construction phase, which is confusing. 

I recompiled everything in devel mode and ran GDB to see if I could find where it goes wrong, and I went all the way down to the calculation step in SideFluxIntegral.C --> "return -MetaPhysicL::raw_value(_diffusion_coef[_qp]) * _grad_u[_qp] * _normals[_qp];". The reported value by _grad_u is wrong. I don't know how to go any deeper than this, so I'm coming to y'all. I've attached my input and geometry files so you can replicate, I got the same bug by using the HeatConduction module to execute. For now I'm going to start checking out older checkpoints of the Master branch to see if I can find the date when the bug begins, but hopefully one of you can find the problem first.

Also attached is a VectorOfPostprocessors VectorPostprocessor C file that I added printing functions to, in order to see the values of the fluxes at every non-linear step, so you can see that the bug isn't in my variable printing.

This is the output of one run:

`Framework Information:
MOOSE Version:           git commit 59fb6b81d3 on 2021-03-23
LibMesh Version:         ed76100f41840c34c146dd0a7f1ff5410370b489
PETSc Version:           3.14.2
SLEPc Version:           3.14.0
Current Time:            Wed Apr 14 12:43:04 2021
Executable Timestamp:    Wed Apr 14 12:42:26 2021

Parallelism:
  Num Processors:          1
  Num Threads:             1

Mesh: 
  Parallel Type:           replicated
  Mesh Dimension:          3
  Spatial Dimension:       3
  Nodes:                   
    Total:                 2205
    Local:                 2205
  Elems:                   
    Total:                 10812
    Local:                 10812
  Num Subdomains:          2
  Num Partitions:          1

Nonlinear System:
  Num DOFs:                2205
  Num Local DOFs:          2205
  Variables:               "temp" 
  Finite Element Types:    "LAGRANGE" 
  Approximation Orders:    "FIRST" 

Execution Information:
  Executioner:             Steady
  Solver Mode:             NEWTON
  MOOSE Preconditioner:    SMP (auto)

FLUXES:
0: -1.32508e-12
1: -2.33283e-12
2: 1.74689e-12
3: -1.63962e-12
4: 1.74689e-12
5: 1.06096e-12
6: -1.73598e-12
7: -1.17187e-12
8: -2.11465e-12
9: -2.22556e-12

Postprocessor Values:
+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+
| time           | heatFlux_0     | heatFlux_1     | heatFlux_2     | heatFlux_3     | heatFlux_4     | heatFlux_5     | heatFlux_6     | heatFlux_7     | heatFlux_8     | heatFlux_9     |
+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+
|   0.000000e+00 |  -1.325077e-12 |  -2.332828e-12 |   1.746893e-12 |  -1.639621e-12 |   1.746893e-12 |   1.060960e-12 |  -1.735984e-12 |  -1.171869e-12 |  -2.114646e-12 |  -2.225555e-12 |
+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+

 0 Nonlinear |R| = 3.863293e+04
FLUXES:
0: 34217.3
1: 28912.7
2: 31062
3: 29094.3
4: 29385.1
5: 31008.6
6: 31068.1
7: 29545.9
8: 29248.2
9: 31292.9
      0 Linear |R| = 3.863293e+04
      1 Linear |R| = 1.008715e+04
      2 Linear |R| = 4.930235e+03
      3 Linear |R| = 3.144405e+03
      4 Linear |R| = 2.255522e+03
      5 Linear |R| = 1.659212e+03
      6 Linear |R| = 1.320328e+03
      7 Linear |R| = 1.088051e+03
      8 Linear |R| = 8.710543e+02
      9 Linear |R| = 7.312698e+02
     10 Linear |R| = 6.073942e+02
     11 Linear |R| = 3.994530e+02
     12 Linear |R| = 1.590399e+02
     13 Linear |R| = 5.332944e+01
     14 Linear |R| = 2.492040e+01
     15 Linear |R| = 1.364132e+01
     16 Linear |R| = 8.115729e+00
     17 Linear |R| = 4.601989e+00
     18 Linear |R| = 2.240354e+00
     19 Linear |R| = 1.385551e+00
     20 Linear |R| = 6.454047e-01
     21 Linear |R| = 3.300779e-01
     22 Linear |R| = 1.610229e-01
     23 Linear |R| = 6.466311e-02
     24 Linear |R| = 2.643870e-02
     25 Linear |R| = 1.540049e-02
     26 Linear |R| = 9.712541e-03
     27 Linear |R| = 5.731784e-03
     28 Linear |R| = 2.833014e-03
     29 Linear |R| = 1.385553e-03
     30 Linear |R| = 6.661644e-04
     31 Linear |R| = 4.195771e-04
     32 Linear |R| = 2.200850e-04
     33 Linear |R| = 1.002690e-04
     34 Linear |R| = 5.212515e-05
     35 Linear |R| = 3.409923e-05
     36 Linear |R| = 2.015706e-05
     37 Linear |R| = 9.101468e-06
     38 Linear |R| = 4.056231e-06
     39 Linear |R| = 2.073032e-06
 1 Nonlinear |R| = 2.073031e-06
FLUXES:
0: 1500
1: 1500
2: 1500
3: 1500
4: 1500
5: 1500
6: 1500
7: 1500
8: 1500
9: 1500
      0 Linear |R| = 2.073031e-06
      1 Linear |R| = 1.444149e-06
      2 Linear |R| = 1.055687e-06
      3 Linear |R| = 8.781906e-07
      4 Linear |R| = 7.564132e-07
      5 Linear |R| = 6.370497e-07
      6 Linear |R| = 4.991177e-07
      7 Linear |R| = 3.870670e-07
      8 Linear |R| = 2.638227e-07
      9 Linear |R| = 1.766717e-07
     10 Linear |R| = 1.157050e-07
     11 Linear |R| = 5.475745e-08
     12 Linear |R| = 2.415633e-08
     13 Linear |R| = 1.517712e-08
     14 Linear |R| = 9.860500e-09
     15 Linear |R| = 6.009218e-09
     16 Linear |R| = 3.067737e-09
     17 Linear |R| = 1.103899e-09
     18 Linear |R| = 4.960291e-10
     19 Linear |R| = 2.666850e-10
     20 Linear |R| = 1.300409e-10
     21 Linear |R| = 5.918693e-11
     22 Linear |R| = 3.057026e-11
     23 Linear |R| = 1.908830e-11
     24 Linear |R| = 1.175909e-11
     25 Linear |R| = 5.782164e-12
     26 Linear |R| = 2.520530e-12
     27 Linear |R| = 9.299019e-13
     28 Linear |R| = 3.708909e-13
     29 Linear |R| = 1.504580e-13
     30 Linear |R| = 5.621707e-14
     31 Linear |R| = 3.076603e-14
     32 Linear |R| = 1.555730e-14
     33 Linear |R| = 7.200368e-15
     34 Linear |R| = 3.160126e-15
     35 Linear |R| = 1.348425e-15
     36 Linear |R| = 7.433555e-16
     37 Linear |R| = 6.012902e-16
     38 Linear |R| = 5.287429e-16
     39 Linear |R| = 4.450518e-16
     40 Linear |R| = 3.703481e-16
     41 Linear |R| = 2.883527e-16
     42 Linear |R| = 1.969545e-16
 2 Nonlinear |R| = 1.474851e-10
FLUXES:
0: 1500
1: 1500
2: 1500
3: 1500
4: 1500
5: 1500
6: 1500
7: 1500
8: 1500
9: 1500
      0 Linear |R| = 1.474851e-10
      1 Linear |R| = 1.004611e-11
      2 Linear |R| = 1.772591e-12
      3 Linear |R| = 6.430570e-13
      4 Linear |R| = 3.221308e-13
      5 Linear |R| = 2.041261e-13
      6 Linear |R| = 1.454971e-13
      7 Linear |R| = 1.160444e-13
      8 Linear |R| = 9.720275e-14
      9 Linear |R| = 8.210715e-14
     10 Linear |R| = 6.696380e-14
     11 Linear |R| = 5.427637e-14
     12 Linear |R| = 4.270465e-14
     13 Linear |R| = 2.910945e-14
     14 Linear |R| = 1.498483e-14
     15 Linear |R| = 7.416829e-15
     16 Linear |R| = 4.706586e-15
     17 Linear |R| = 2.946539e-15
     18 Linear |R| = 1.823344e-15
     19 Linear |R| = 9.573813e-16
     20 Linear |R| = 3.630267e-16
     21 Linear |R| = 1.537032e-16
     22 Linear |R| = 7.623668e-17
     23 Linear |R| = 4.178874e-17
     24 Linear |R| = 2.436014e-17
     25 Linear |R| = 1.166257e-17
     26 Linear |R| = 6.358350e-18
     27 Linear |R| = 4.137238e-18
     28 Linear |R| = 2.242834e-18
     29 Linear |R| = 8.246964e-19
     30 Linear |R| = 2.948909e-19
     31 Linear |R| = 1.449013e-19
     32 Linear |R| = 5.504189e-20
     33 Linear |R| = 2.524015e-20
     34 Linear |R| = 1.519536e-20
     35 Linear |R| = 9.718212e-21
 3 Nonlinear |R| = 1.398896e-10
 Solve Converged!

Postprocessor Values:
+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+
| time           | heatFlux_0     | heatFlux_1     | heatFlux_2     | heatFlux_3     | heatFlux_4     | heatFlux_5     | heatFlux_6     | heatFlux_7     | heatFlux_8     | heatFlux_9     |
+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+
|   0.000000e+00 |  -1.325077e-12 |  -2.332828e-12 |   1.746893e-12 |  -1.639621e-12 |   1.746893e-12 |   1.060960e-12 |  -1.735984e-12 |  -1.171869e-12 |  -2.114646e-12 |  -2.225555e-12 |
|   1.000000e+00 |   1.500000e+03 |   1.500000e+03 |   1.500000e+03 |   1.500000e+03 |   1.500000e+03 |   1.500000e+03 |   1.500000e+03 |   1.500000e+03 |   1.500000e+03 |   1.500000e+03 |
+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+----------------+

FLUXES:
0: 1500
1: 1500
2: 1500
3: 1500
4: 1500
5: 1500
6: 1500
7: 1500
8: 1500
9: 1500
'

Thanks y'all!

[FluxBugInputs.zip](https://github.com/idaholab/moose/files/6312404/FluxBugInputs.zip)

**Comment by wrkendrick** [comment, marked as answer]
SOLVED: Turns out an initial condition was set too high and on the first calculation it would (correctly) calculate a huge heat flux, and after 1 linear iteration it would fix itself.

**Reply by GiudGiud** [reply]
nice Sterling and I were wondering what to do about this yesterday
